File: helfomat-core/src/main/java/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditTeaserImageEvent.java
package de.helfenkannjeder.helfomat.core.organization.event;

import de.helfenkannjeder.helfomat.core.organization.Organization;
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;
import de.helfenkannjeder.helfomat.core.picture.PictureId;

/**
 * @author Valentin Zickner
 */
@SuppressWarnings({"WeakerAccess", "CanBeFinal"})
public class OrganizationEditTeaserImageEvent extends OrganizationEditEvent {
    private PictureId teaserImage;

    protected OrganizationEditTeaserImageEvent() {
    }

    public OrganizationEditTeaserImageEvent(OrganizationId organizationId, PictureId teaserImage) {
        super(organizationId);
        this.teaserImage = teaserImage;
    }

    public PictureId getTeaserImage() {
        return teaserImage;
    }

    @Override
    public Organization.Builder applyOnOrganizationBuilder(Organization.Builder organization) {
        return organization.setTeaserImage(teaserImage);
    }

    @Override
    public <T> T visit(OrganizationEventVisitor<T> visitor) {
        return visitor.visit(this);
    }

    @Override
    public String toString() {
        return "OrganizationEditTeaserImageEvent{" +
            "teaserImage=" + teaserImage +
            '}';
    }
}

File: helfomat-core/src/main/kotlin/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditTeaserImageEvent.kt
package de.helfenkannjeder.helfomat.core.organization.event

import de.helfenkannjeder.helfomat.core.organization.Organization
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor
import de.helfenkannjeder.helfomat.core.organization.OrganizationId
import de.helfenkannjeder.helfomat.core.picture.PictureId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditTeaserImageEvent(
    override val organizationId: OrganizationId,
    val teaserImage: PictureId?
) : OrganizationEditEvent(organizationId) {

    override fun applyOnOrganizationBuilder(organizationBuilder: Organization.Builder?, strictMode: Boolean): Organization.Builder? {
        return organizationBuilder?.setTeaserImage(teaserImage)
    }

    override fun <T> visit(visitor: OrganizationEventVisitor<T>): T {
        return visitor.visit(this)
    }

}